controllers/operator/manager.go

```diff
@@ -164,7 +164,11 @@ func (m *ODLMOperator) GetCatalogSourceFromPackage(ctx context.Context, packageN
 			if !channelCheck(channel, pm.Status.Channels) || (excludedCatalogSources != nil && util.Contains(excludedCatalogSources, pm.Status.CatalogSource)) {
 				continue
 			}
-			catalogSourceCandidate = append(catalogSourceCandidate, CatalogSource{Name: pm.Status.CatalogSource, Namespace: pm.Status.CatalogSourceNamespace, OpNamespace: namespace, RegistryNamespace: registryNs})
+			catalogsource := &olmv1alpha1.CatalogSource{}
+			if err := m.Reader.Get(ctx, types.NamespacedName{Name: pm.Status.CatalogSource, Namespace: pm.Status.CatalogSourceNamespace}, catalogsource); err != nil {
+				klog.Warning(err)
```

Comment:
I thought we could return error here.
Usually when `err != nil`, we will get `catalogsource` as `nil`. Therefore, it will error out in the next line when we try to read `catalogsource.Spec.Priority`